spec/unit/backtracie_spec.rb

```diff
@@ -271,6 +271,118 @@ class ClassWithBlockInsideMethodDefinedUsingDefinedMethod
       end
     end
 
+    context "when sampling a tracepoint hitting define_method defined inside a block" do
+
+      # This is a minimal (yes, believe it or not, I couldn't get it any smaller) reproduction of an issue I hit in
```

Comment:
Yeah, this is a weird one. Only simplification I can think of is using an actual tracepoint -- `TracePoint.new(:c_call)` instead of the more generic `set_trace_func` maybe?